pkg/operator/finalizer.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	monitoringclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+)
+
+// SyncFinalizers adds or removes the finalizer form the workload resource(Prometheus, PrometheusAgent, Alertmanager and ThanosRuler).
+// It returns true if the finalizers were modified, otherwise false. The second return value is an error, if any.
```

Comment:
(nit)
```suggestion
// Adds or removes the `monitoring.coreos.com/status-cleanup` finalizer to or from the object's list of finalizers. It returns true if the object's list of finalizers was modified, otherwise false. Removes the object from the reconciliation tracker if its deletion is in progress. The second return value is an error, if any.
```
The comment would change as we update this function, but for now this is good.

Reply:
We can probably make it clear that this function is only intended for Prometheus, Alertmanager, ThanosRuler and PrometheusAgent objects. But this should be good for starters.

Reply:
done